File: sangjinHan/월간코테/2.py
def solution(s):
    zero = 0
    count = 0
    answer = []
    
    while True:
        if s == "1":
            break
        
        temp = ""
        binary = ""
        
        # 0 제거
        for i in s:
            if i == "0":
                zero += 1
            else:
                temp += "1"
        
        # 제거 후 길이
        length = len(temp)
        
        # 이진 변환 결과
        while True:
            if length == 0 :
                break
            binary += str(length%2)
            length = length//2
        
        binary = int(binary[::-1])
        
        s = str(binary)
        count += 1
    
    answer.append(count)
    answer.append(zero)
    
    return answer

File: sangjinHan/Study/13_1/2.swift
import Foundation

import Foundation

func solution(_ encrypted_text:String, _ key:String, _ rotation:Int) -> String {
    var asciiKey = Array(key).map{ Int($0.asciiValue!)-96 }
    let encypt = Array(encrypted_text).map{ Int($0.asciiValue!) }

    var encodedStr = [Int]()

    let rotatedStr = rotate(encypt,rotation)


    for (index,char) in rotatedStr.enumerated() {

        let newChar = (char - asciiKey[index]) < 97 ? (char - asciiKey[index]) + 26 : (char - asciiKey[index])

        encodedStr.append(newChar)
    }

    let answer = encodedStr.map{ String(UnicodeScalar($0)!) }.joined()

    return answer
}

func rotate(_ arr: [Int], _ num: Int) -> [Int] {
    var returnArr = arr

    var shotNum = num % arr.count

    if shotNum < 0 {
        let newNum = -shotNum
        var slice = Array(returnArr[(arr.count-newNum)...])

        returnArr.removeSubrange((arr.count-newNum)...)
        slice += returnArr

        returnArr = slice
    } else if shotNum > 0 {
        let slice = Array(returnArr[0..<shotNum])
    
        returnArr.removeSubrange(0..<shotNum)
        returnArr += slice
    }

    return returnArr
}

// print(solution("asdqweqwe","abcdefghi", 10))

//	"qyyigoptvfb","abcdefghijk", 3

